images/metering-ansible-operator/roles/meteringconfig/tasks/configure_tls.yml

```diff
@@ -1,7 +1,24 @@
 ---
 
-- name: Configure TLS and client-side authentication in Presto
-  include_tasks: configure_presto_tls.yml
+- name: Configure TLS
+  block:
+  - name: Create temporary directory to store all the necessary certificates/keys
+    tempfile:
+      suffix: certificates
+      state: directory
+    register: certificates_dir
 
-- name: Configure TLS and authentication in the reporting-operator
-  include_tasks: configure_reporting_operator_tls.yml
+  - name: Generate root certificate authority
+    include_tasks: configure_root_ca.yml
+
+  - name: Configure TLS and client-side authentication in Presto
+    include_tasks: configure_presto_tls.yml
+
+  - name: Configure TLS and authentication in the reporting-operator
+    include_tasks: configure_reporting_operator_tls.yml
+
+  always:
+  - name: Cleanup the temporary directory which held the certificates and keys
+    file:
+      path: "{{ certificates_dir.path }}"
```

Comment:
Since `certificates_dir` is in the block, does the scope here work? I guess since it's not evaluated until later it might be okay?

Reply:
Scoping should be a problem as I also do this:
```
  vars:
    root_ca_secret_exists: "{{ root_ca_secret_buf and root_ca_secret_buf.resources | length > 0 }}"
  when: meteringconfig_tls_enabled and not root_ca_secret_exists
```

which evaluates properly